test/addons-napi/test_promise/test.js

```diff
@@ -44,7 +46,12 @@ const test_promise = require(`./build/${common.buildType}/test_promise`);
 }
 
 assert.strictEqual(test_promise.isPromise(test_promise.createPromise()), true);
-assert.strictEqual(test_promise.isPromise(Promise.reject(-1)), true);
+
+const rejectPromise = Promise.reject(-1);
+assert.strictEqual(test_promise.isPromise(rejectPromise), true);
+//Dummy rejection handler
+rejectPromise.catch(() => {});
```

Comment:
would it maybe make sense to test the error object in the catch?

Reply:
Sure! I could make the change. Would it be a good start by testing the equality between the reason code input of static rejection and the error code inside the catch?

Reply:
@babygoat I think there's no rejection error code here - the rejection value itself is `-1`, I think, so it's a number and can't have properties of its own. We can test that the value is `-1` here, though :)

Reply:
Hello @addaleax, thanks for your explanation! Yeah, you are right the rejection value is not an error object itself but a pure number (-1). But your suggestion is exactly the way I'm currently planning to do! Maybe I should provide the pseudo code for the implementation instead of only vague proposal. I rebase and upload the new commit for the further discussion